src/common.h

```diff
@@ -103,8 +99,13 @@ enum class Result {
   Error,
 };
 
-#define WABT_SUCCEEDED(x) ((x) == ::wabt::Result::Ok)
-#define WABT_FAILED(x) ((x) == ::wabt::Result::Error)
+template<typename T>
+void ZeroMemory(T& v) {
+  memset(&v, 0, sizeof(v));
+}
+
+inline bool Succeeded(Result result) { return result == Result::Ok; }
```

Comment:
Honestly, if we're refactoring this, I'd rather make it so that we could just `Result` directly as a bool.